nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ListenSMTP.java

```diff
@@ -31,101 +30,75 @@
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
 import org.apache.nifi.ssl.SSLContextService;
-
-import org.apache.nifi.processors.email.smtp.event.SmtpEvent;
-import org.apache.nifi.processors.email.smtp.handler.SMTPResultCode;
-import org.apache.nifi.processors.email.smtp.handler.SMTPMessageHandlerFactory;
+import org.springframework.util.StringUtils;
+import org.subethamail.smtp.MessageContext;
+import org.subethamail.smtp.server.SMTPServer;
 
 @Tags({"listen", "email", "smtp"})
 @InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
-@CapabilityDescription("This processor implements a lightweight SMTP server to an arbitrary port, " +
-        "allowing nifi to listen for incoming email. " +
-        "" +
-        "Note this server does not perform any email validation. If direct exposure to the internet is sought," +
-        "it may be a better idea to use the combination of NiFi and an industrial scale MTA (e.g. Postfix)")
+@CapabilityDescription("This processor implements a lightweight SMTP server to an arbitrary port, "
+        + "allowing nifi to listen for incoming email. Note this server does not perform any email "
+        + "validation. If direct exposure to the internet is sought, it may be a better idea to use "
+        + "the combination of NiFi and an industrial scale MTA (e.g. Postfix)")
 @WritesAttributes({
-        @WritesAttribute(attribute = "mime.type", description = "The value used during HELO"),
-        @WritesAttribute(attribute = "smtp.helo", description = "The value used during HELO"),
-        @WritesAttribute(attribute = "smtp.certificates.*.serial", description = "The serial numbers for each of the " +
-                "certificates used by an TLS peer"),
-        @WritesAttribute(attribute = "smtp.certificates.*.principal", description = "The principal for each of the " +
-                "certificates used by an TLS peer"),
-        @WritesAttribute(attribute = "smtp.from", description = "The value used during MAIL FROM (i.e. envelope)"),
-        @WritesAttribute(attribute = "smtp.to", description = "The value used during RCPT TO (i.e. envelope)"),
-        @WritesAttribute(attribute = "smtp.src", description = "The source IP of the SMTP connection")})
-
-public class ListenSMTP extends AbstractProcessor {
-    public static final String SMTP_HELO = "smtp.helo";
-    public static final String SMTP_FROM = "smtp.from";
-    public static final String SMTP_TO = "smtp.to";
-    public static final String MIME_TYPE = "message/rfc822";
-    public static final String SMTP_SRC_IP = "smtp.src";
-
-
-    protected static final PropertyDescriptor SMTP_PORT = new PropertyDescriptor.Builder()
+    @WritesAttribute(attribute = "smtp.helo", description = "The value used during HELO"),
```

Comment:
@olegz My apologies but I missed the mime type completely...

The original description is wrong but in any case:

This is NOT the mime type of the attachments (these are extracted by ExtractEmailAttachments) 
Instead this was set with a non variable value "message/rfc822" (This is the standard mime type of a text file containing an email message (which is what this Listener produces)).

As with previous the addition of mime.type follows the pattern used everywhere around as seen here:

https://github.com/apache/nifi/search?utf8=%E2%9C%93&q=%40WritesAttribute%28attribute+%3D+%22mime.type%22


Apologies for noting this earlier.